doozerlib/rpmcfg.py

```diff
@@ -479,6 +512,51 @@ def wait(n):
             self.post_build(scratch)
             return self.distgit_key, self.build_status
 
+    target_golang_versions = {}  # a dict of cached target -> golang_version mappings
+    target_golang_versions_lock = threading.Lock()
+
+    def assert_golang_versions(self):
+        """ If this RPM component has multiple targets, check if all buildroots have consistent versions of golang compilers
+        """
+        check_mode = self.runtime.group_config.check_golang_versions or "x.y"  # no: do not check; x.y: only major and minor version; exact: the z-version must be the same
+        if check_mode == "no":
+            return
+        # target_golang_versions = {}  # a dict of target -> golang_version mappings
+        golang_components = ["golang", "golang-scl-shim"]
```

Comment:
i think you may want to add 
```suggestion
        golang_components = ["golang", "golang-scl-shim", "go-toolset"]
```

in el8 buildroots the RPM that actually gets installed and used is `go-toolset`. `golang` will probably be the same version but i'd rather check directly. i'm not certain this is the right thing to do though, maybe try it out and see. (we had a great testbed for golang mismatches in 4.4 and 4.5 until yesterday they were trued up...)

Reply:
Do we rebuild `golang-scl-shim` for every `go-toolset` z version update? If no I think it is ok for golang-scl-shim to have differing z version...
For `go-toolset`, it seems to me the package name is `go-toolset-1.y` where y is `{12,13,14,15,...}` Don't really want to hardcode the y versions here. Seems I need to fetch the latest builds for all components with the `-build` tag then filter those prefixed with `go-toolset-*`...

Reply:
We don't rebuild `golang-scl-shim`, no. It might make sense to do that in the future for clarity, but I don't think we want to count on it yet. Every time we do that we have to ask someone to tag it in... that might also change though...

Filtering by prefix instead of using pkg name seems like a good strategy. Where we use inheritance to include go-toolset, we can have more than one minor version available; that's why golang-scl-shim hardcodes a specific version. E.g.:
```
$ brew latest-build rhaos-4.3-rhel-7-build go-toolset-1.1{0,1,2,3}
Build                                     Tag                   Built by
----------------------------------------  --------------------  ----------------
go-toolset-1.10-1.10.3-7.el7              devtools-2018.4-rhel-7  deparker
go-toolset-1.11-1.11.13-1.el7             devtools-2019.3-rhel-7  tstellar
go-toolset-1.12-1.12.12-4.el7             devtools-2019.4-rhel-7-candidate  deparker
```
For 4.4+ I have asked to just tag the specific builds in rather than inherit, but we haven't gone back to previous versions and done the same (probably won't unless there's some need for a change).

Reply:
Thanks for the clarification. I think the logic should be:
- get the `x.y` version of `golang-scl-shim`
- fetch the latest version of `go-toolset-${x}-${y}` where `x` and `y` are from `golang-scl-shim`.
- check whether the `x.y.z` version of `go-toolset-${x}-${y}` matches the one for RHEL8.
I'll update the PR to reflect the above logic.

Reply:
PR Updated